### PR TITLE
Support signed byte and ushort vertex buffer attribute types

### PR DESCRIPTION
**Pull request by Alex-MSFT**
This PR adds support for signed byte and ushort vertex buffer attribute types.  This has been tested on Mac OSX, Android, iOS, and Windows via D3D11, which are what is needed for BabylonNative.  Other formats are not fully tested as of this change.

This is one part of addressing https://github.com/BabylonJS/BabylonNative/issues/729

### CODE REVIEW

**Review by ryantrem — APPROVED** [review]

File: include/bgfx/c99/bgfx.h [hunk 1]
```diff
@@ -145,11 +145,13 @@ typedef enum bgfx_attrib
  */
 typedef enum bgfx_attrib_type
 {
-    BGFX_ATTRIB_TYPE_UINT8,                   /** ( 0) Uint8                          */
-    BGFX_ATTRIB_TYPE_UINT10,                  /** ( 1) Uint10, availability depends on: `BGFX_CAPS_VERTEX_ATTRIB_UINT10`. */
-    BGFX_ATTRIB_TYPE_INT16,                   /** ( 2) Int16                          */
-    BGFX_ATTRIB_TYPE_HALF,                    /** ( 3) Half, availability depends on: `BGFX_CAPS_VERTEX_ATTRIB_HALF`. */
-    BGFX_ATTRIB_TYPE_FLOAT,                   /** ( 4) Float                          */
+    BGFX_ATTRIB_TYPE_INT8,                    /** ( 0) Int8, availability depends on: `BGFX_CAPS_VERTEX_ATTRIB_INT8`. */
```
Comment (bghgary) [inline include/bgfx/c99/bgfx.h:148]: Are we 100% sure this enum is not serialized anywhere?

**Review by bghgary — APPROVED** [review]

File: include/bgfx/c99/bgfx.h (hunk 1 above)
Comment (bghgary) [reply include/bgfx/c99/bgfx.h:148]: Would it be safer to add the new enums to the end?

Comment (Alex-MSFT) [reply include/bgfx/c99/bgfx.h:148]: Assignment of IDs happens in vertexlayout.cpp, see s_attribTypeToId

Comment (bghgary) [reply include/bgfx/c99/bgfx.h:148]: Yes, I see that. I'm just not sure this is used directly in bgfx's shader compiler or something.

**Pull request merged**